### PR TITLE
fix: references must be unique

### LINKED ISSUE #593: BUG - `references` contains the same namespace multiple times
### Description
According to [the spec](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Toc406397932), the same namespace must not be included multiple times:

> 3.4.1 Attribute Namespace
>
>The [edmx:Include](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Element_edmx:Include) element MUST provide a [Namespace](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Namespace) value for the Namespace attribute. The value MUST match the namespace of a schema defined in the referenced CSDL document. The same namespace MUST NOT be included more than once, even if it is declared in more than one referenced document.

This constraint is violated by the annotation converter, leading to multiple aliases mapping to the same namespace, i.e. the mapping is not invertible.

### PR DESCRIPTION
**Pull request by drktfl**
See http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Toc406397932

Fixes #593

### CODE REVIEW

File: packages/annotation-converter/src/converter.ts [hunk 1]
```diff
@@ -1541,6 +1542,40 @@ function convertTypeDefinition(converter: Converter, rawTypeDefinition: RawTypeD
     return convertedTypeDefinition;
 }
 
+function getReferences(rawMetadataReferences: Reference[]) {
+    if (rawMetadataReferences.length === 0) {
+        return VocabularyReferences;
+    }
+
+    // the provided references must be unique
```
Comment (drktfl) [inline packages/annotation-converter/src/converter.ts:1550]: this check is probably a bit paranoid (O(n^2)) given that this was not checked at all before?

**Comment by nlunets** [comment]
Reading the spec i have a doubt.

The same namespace must not be used multiple times -> probably for different document includes but not if it's the same document that gets included multiple times.
The same alias must not be used multiple times with different namespace -> OK
Different alias may not point to the same namespace -> i don't see that one 

So i am not sure that all the checks are necessary, what do you think ?

**Comment by drktfl** [comment]
Hmm, this is how I interpreted http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Element_edmx:Reference, but maybe I missed something? This is my interpretation:

> 3.3.1 Attribute Uri
>
>The [edmx:Reference](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Element_edmx:Reference) element MUST specify a Uri attribute. The Uri attribute uniquely identifies a model, so **two references MUST NOT specify the same URI**.

So technically there must not be two `edmx:Reference` with the same Uri - but we don't check that in the converter, because when parsing, the "one reference - multiple includes" is multiplied out into a flat list. So the information whether the same Uri was originally included only once is lost (and it does not matter that much, I think).

&rarr; do not check whether `uri` is unique

> 3.4.1 Attribute Namespace
>
> The [edmx:Include](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Element_edmx:Include) element MUST provide a [Namespace](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Namespace) value for the Namespace attribute. The value MUST match the namespace of a schema defined in the referenced CSDL document. **The same namespace MUST NOT be included more than once, even if it is declared in more than one referenced document.**

A namespace may only be included once, no matter where it comes from. I read that in a way that even this is not allowed:
```xml
<edmx:Reference Uri="https://example.org/namespace/1">
    <edmx:Include Alias="X" Namespace="org.example.namespace.1"/>
    <edmx:Include Alias="Y" Namespace="org.example.namespace.1"/>
</edmx:Reference>
```
because the same namespace _is_ `<edxm:Include>`d twice.

```xml
<edmx:Reference Uri="https://example.org/namespace/1">
    <edmx:Include Alias="X" Namespace="org.example.namespace.1"/>
</edmx:Reference>
<edmx:Reference Uri="https://example.org/namespace/2">
    <edmx:Include Alias="Y" Namespace="org.example.namespace.1"/>
</edmx:Reference>
```
is not allowed either

&rarr; check whether `namespace` is unique

> Aliases are document-global, so [edmx:Include](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_Element_edmx:Include) and [edm:Schema](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/part3-csdl/odata-v4.0-errata02-os-part3-csdl-complete.html#_The_edm:Schema_Element) **elements within a document MUST NOT assign the same alias to different namespaces** and MUST NOT specify an alias with the same name as an in-scope namespace.

&rarr; check whether `alias` is unique (this is trivial as otherwise one could not resolve the namespace from the alias)

Do I miss a case?

**Comment by drktfl** [comment]
Bottom line: alias and namespace should be 1:1

**Comment by nlunets** [comment]
Ah i missed the part where Uri should be unique :)
Then i guess this makes sense

File: packages/annotation-converter/src/converter.ts (hunk 1 above)
Comment (nlunets) [reply packages/annotation-converter/src/converter.ts:1550]: Hmmm maybe not throwing but maybe we should keep it for the diagnostics, what do you think ?

Comment (drktfl) [reply packages/annotation-converter/src/converter.ts]: Makes sense. This no longer throws now

**Review by nlunets — APPROVED** [review]

**Pull request merged**